homeassistant/components/zwave_js/cover.py

```diff
@@ -402,6 +411,28 @@ def _tilt_range(self) -> int:
         """Return range of valid tilt positions."""
         return abs(SlatStates.CLOSED_2 - SlatStates.CLOSED_1)
 
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Open the cover."""
+        if self._up_value:
+            await self._async_set_value(self._up_value, True)
+        else:
+            await super().async_open_cover(**kwargs)
```

Comment:
When will these untested cases happen? Can we test them for a known device case (fixture)?

Reply:
They shouldn't, really. Window Covering CC always has the up/down values.
Can we assert somehow that `get_zwave_value` does not return None?

I don't think the discovery schema lets me require that the up/down values exist with the same property key as the primary value, does it?

Reply:
No, I don't think the discovery schema relays type information like that. We can cast the value to the correct type with `typing.cast` in the init method.

Reply:
This one needs updating too.

Reply:
oops